lisa/wlgen/rta.py

```diff
@@ -592,12 +610,38 @@ def __init__(self, start_pct=0, end_pct=100, delta_pct=10, time_s=1,
         # clip the last step
         steps[-1] = end_pct
 
+        # Clamp values for previous phases.
+        # These values are remembered so the clamp values are not set on the start
+        #   of every phase
+        # The initial values should probably be set on the first phase,
+        #   this protects against retaining a clamp from a previous loop.
+        clamp_min = None
+        clamp_max = None
+
         phases = []
-        for load in steps:
+        for idx, load in enumerate(steps):
+            # If clamp values are unchanged, don't bother setting them again
+            new_clamp_min = clamp_mins(idx, load)
+            if new_clamp_min == clamp_min:
+                new_clamp_min = None
+            new_clamp_max = clamp_maxs(idx, load)
+            if new_clamp_max == clamp_max:
+                new_clamp_max = None
+                
             if load == 0:
-                phase = Phase(time_s, 0, 0, cpus)
+                phase = Phase(time_s, 0, 0, cpus, 
+                              clamp_min=new_clamp_min, 
+                              clamp_max=new_clamp_max)
             else:
-                phase = Phase(time_s, period_ms, load, cpus)
+                phase = Phase(time_s, period_ms, load, cpus, 
+                              clamp_min=new_clamp_min, 
+                              clamp_max=new_clamp_max)
+                
+            if new_clamp_min != None:
```

Comment:
Use `new_clamp_min is not None` for comparison with None. None is a singleton so it will always work, and using `is/is not` is recommended by PEP8 in these cases (mostly for speed purpose and to avoid broken equality implementations I assume):
https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
This change has now been made.